metron-platform/metron-data-management/src/main/java/org/apache/metron/dataloads/extractor/TransformFilterExtractorDecorator.java

```diff
@@ -0,0 +1,236 @@
+    }
+    zkClient = setupClient(zkClient, zkClientUrl);
+    this.globalConfig = getGlobalConfig(zkClient);
+    this.stellarContext = createContext(zkClient);
+    StellarFunctions.initialize(stellarContext);
+    this.transformProcessor = new StellarProcessor();
+    this.filterProcessor = new StellarPredicateProcessor();
+  }
+
+  private String getFilter(Map<String, Object> config, String valueFilter) {
+    return ConversionUtils.convertOrFail(config.get(valueFilter), String.class);
+  }
+
+  /**
+   * Get a map of the transformations from the config of the specified type
+   * @param config main config map
+   * @param type the transformation type to get from config
+   * @return map of transformations.
+   */
+  private Map<String, String> getTransforms(Map<String, Object> config, String type) {
+    Map<Object, Object> transformsConfig = ConversionUtils.convertOrFail(config.get(type), Map.class);
+    Map<String, String> transforms = new LinkedHashMap<>();
+    for (Map.Entry<Object, Object> e : transformsConfig.entrySet()) {
+      String key = ConversionUtils.convertOrFail(e.getKey(), String.class);
+      String val = ConversionUtils.convertOrFail(e.getValue(), String.class);
+      transforms.put(key, val);
+    }
+    return transforms;
+  }
+
+  /**
+   * Creates a Zookeeper client if it doesn't exist and a url for zk is provided.
+   * @param zookeeperUrl The Zookeeper URL.
+   */
+  private Optional<CuratorFramework> setupClient(Optional<CuratorFramework> zkClient, String zookeeperUrl) {
+    // can only create client if we have a valid zookeeper URL
+    if (!zkClient.isPresent()) {
+      if (StringUtils.isNotBlank(zookeeperUrl)) {
+        CuratorFramework client = ConfigurationsUtils.getClient(zookeeperUrl);
+        client.start();
+        return Optional.of(client);
+      } else {
+        LOG.warn("Unable to setup zookeeper client - zk_quorum url not provided. **This will limit some Stellar functionality**");
+        return Optional.empty();
+      }
+    } else {
+      return zkClient;
+    }
+  }
+
+  private Map<String, Object> getGlobalConfig(Optional<CuratorFramework> zkClient) {
+    if (zkClient.isPresent()) {
+      try {
+        return JSONUtils.INSTANCE.load(
+                new ByteArrayInputStream(ConfigurationsUtils.readGlobalConfigBytesFromZookeeper(zkClient.get())),
+                new TypeReference<Map<String, Object>>() {
+                });
+      } catch (Exception e) {
+        LOG.warn("Exception thrown while attempting to get global config from Zookeeper.", e);
+      }
+    }
+    return new LinkedHashMap<>();
+  }
+
+  private Context createContext(Optional<CuratorFramework> zkClient) {
+    Context.Builder builder = new Context.Builder();
+    if (zkClient.isPresent()) {
+      builder.with(Context.Capabilities.ZOOKEEPER_CLIENT, zkClient::get)
+              .with(Context.Capabilities.GLOBAL_CONFIG, () -> globalConfig);
```

Comment:
By empty, do you mean null or "{}"? Does Stellar handle that differently from choosing not to add the capability at all?